### PR TITLE
ncurses: build without Ada on Linux

### PR DESCRIPTION
**Pull request by lexicol**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/ncurses.rb [hunk 1]
```diff
@@ -23,14 +23,24 @@ class Ncurses < Formula
...
+      --enable-widec
+      --with-shared
+      --with-gpm=no
+    ]
+
+    on_linux do
+      # Homebrew GCC does not support Ada
+      # https://github.com/Homebrew/linuxbrew-core/pull/2841
+      args << "--without-ada"
+    end
```
Comment (carlocab) [inline Formula/ncurses.rb:41]: I want to try to preserve `git blame` here. Can you just put
```ruby
args = []
on_linux do
  # Homebrew GCC does not support Ada
  # https://github.com/Homebrew/linuxbrew-core/pull/2841
  args << "--without-ada"
end
```
at the start of the formula and then add `*args` as the last argument in the list, so you don't need to refactor them all into an array?

**Review by carlocab — CHANGES_REQUESTED** [review]
See above. Thanks!

**Comment by lexicol** [comment]
I wonder, would it make sense to disable Ada bindings on macOS too?

**Comment by carlocab** [comment]
Sure; I guess it doesn't hurt.

**Comment by carlocab** [comment]
Merged in https://github.com/Homebrew/homebrew-core/pull/74319

**Pull request closed without merging**